packages/mobile/src/fees/reducer.ts

```diff
@@ -1,51 +1,82 @@
-import { combineReducers } from 'redux'
-import { Actions, ActionTypes, FeeType } from 'src/fees/actions'
+import { createAction, createReducer } from '@reduxjs/toolkit'
+import { FeeInfo } from 'src/fees/saga'
+import { getRehydratePayload, REHYDRATE, RehydrateAction } from 'src/redux/persist-helper'
 
-interface FeeEstimateState {
-  feeInWei: string | null
-  lastUpdated: number | null
+export enum FeeType {
+  INVITE = 'invite',
+  SEND = 'send',
+  EXCHANGE = 'exchange',
+  RECLAIM_ESCROW = 'reclaim-escrow',
+  REGISTER_DEK = 'register-dek',
 }
 
-const feeEstimateInitialState = {
-  feeInWei: null,
-  lastUpdated: null,
+export interface FeeEstimateState {
+  usdFee: string | null
+  lastUpdated: number
+  loading: boolean
+  error: boolean
+  feeInfo?: FeeInfo
 }
 
-function createEstimateReducer(feeType: FeeType) {
-  return function estimateReducer(
-    state: FeeEstimateState = feeEstimateInitialState,
-    action: ActionTypes
-  ): FeeEstimateState {
-    if (action.feeType !== feeType) {
-      return state
-    }
-
-    switch (action.type) {
-      case Actions.FEE_ESTIMATED:
-        return {
-          ...state,
-          feeInWei: action.feeInWei,
-        }
-      default:
-        return state
+export interface State {
+  estimates: {
+    [tokenAddress: string]: {
+      [feeType in FeeType]: FeeEstimateState | undefined
     }
   }
 }
 
-export interface State {
-  estimates: {
-    invite: FeeEstimateState
-    send: FeeEstimateState
-    exchange: FeeEstimateState
-    reclaimEscrow: FeeEstimateState
-  }
+export const initialState: State = {
+  estimates: {},
 }
 
-const estimatesReducer = combineReducers({
-  invite: createEstimateReducer(FeeType.INVITE),
-  send: createEstimateReducer(FeeType.SEND),
-  exchange: createEstimateReducer(FeeType.EXCHANGE),
-  reclaimEscrow: createEstimateReducer(FeeType.RECLAIM_ESCROW),
-})
+export interface EstimateFeeAction {
+  feeType: FeeType
+  tokenAddress: string
+  paymentID?: string // Must be set if feeType === RECLAIM_ESCROW
+}
 
-export const reducer = combineReducers({ estimates: estimatesReducer })
+const rehydrate = createAction<any>(REHYDRATE)
+export const estimateFee = createAction<EstimateFeeAction>('FEES/ESTIMATE_FEE')
+export const feeEstimated = createAction<{
+  tokenAddress: string
+  feeType: FeeType
+  estimation: FeeEstimateState
+}>('FEES/FEE_ESTIMATED')
+
+export const reducer = createReducer(initialState, (builder) => {
```

Comment:
Fine for now, but in the future we should favor `createSlice` instead.
See https://github.com/valora-inc/wallet/issues/1442

Reply:
Forgot about this 😓  Will keep in mind